code/src/components/MovieDetail.js

```diff
@@ -0,0 +1,76 @@
+import React, { useEffect, useState } from "react";
+import { useParams } from "react-router-dom";
+import Moment from "react-moment";
+import { API_KEY } from "Urls";
+import Loading from "../Loading";
+import { Link } from "react-router-dom";
+import page404 from "../images/page404.mp4";
+
+const MovieDetail = () => {
+  const [movie, setMovie] = useState([]);
+  const [loading, setLoading] = useState(false);
+  const { id } = useParams();
+  const [hasError, setHasError] = useState(false);
+
+  useEffect(() => {
+    setLoading(true);
+    fetch(`https://api.themoviedb.org/3/movie/${id}?api_key=${API_KEY}&language=en-US`)
+      .then((res) => res.json())
+      .then((json) => {
+        if (json.id) {
+          setTimeout(() => setLoading(false), 2000);
+          setMovie(json);
+        } else {
+          setHasError(true);
+        }
+      });
+  }, [id]);
+
+  if (hasError) {
+    return (
+      <div className="pageNotFoundContainer">
+        <div className="imgPageNotFound">
+          <video autoPlay loop muted src={page404} alt={"no page found"} />
+        </div>
+
+        <div className="textPageNotFound">
+          <h2> Sorry page not found...</h2>
+        </div>
+        <div className="buttonPageNotFound">
+          <button>
+            <Link className="linkPageNotFound" to="/">
+              Go back
+            </Link>
+          </button>
+        </div>
+      </div>
+    );
+  }
+
+  return (
+    <div>
+      {loading && <Loading />}
+      {movie && (
+        <div className="movieDetalisBox">
+          <img className="backgroundMovieDetails" src={`http://image.tmdb.org/t/p/w1280${movie.backdrop_path}`} alt={movie.id} />
+          <div className="MovieDetailsInfo">
```

Comment:
Those two error messages are showing up in the devtools:
![grafik](https://user-images.githubusercontent.com/77005046/140902481-edd9efe1-8efd-44c8-b77a-67c220cabd0d.png)
They are connected to line 55 and 57 of your code, because on mounting the page, the data isn't there to display yet. It will be fetched for the first time only after the mounting. So what you can do is to check, if the data is there first, like so:
```
src={
          movieDetails.backdrop_path
            ? `https://image.tmdb.org/t/p/w1280${movieDetails.backdrop_path}`
            : ""
        }
```